packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -405,88 +1824,104 @@ class AdapterClass extends EventEmitter {
      *     }
      * </code></pre>
 
-     * @alias supportsFeature
-     * @memberof Adapter
-     * @param {string} featureName the name of the feature to check
-     * @returns {boolean} true/false if the feature is in the list of supported features
+     * @param featureName the name of the feature to check
+     * @returns true/false if the feature is in the list of supported features
      */
-    supportsFeature(featureName) {
-        return SUPPORTED_FEATURES.includes(featureName);
+    supportsFeature(featureName: unknown): boolean {
+        if (typeof featureName === 'string') {
+            return SUPPORTED_FEATURES.includes(featureName);
+        } else {
+            return false;
+        }
     }
```

Comment:
If we define `SUPPORTED_FEATURES` `as const`, and add a type `type Feature = SUPPORTED_FEATURES[number]`, we don't need a separate overload for this and the following is enough (and nicer for the IDE):

```suggestion
    supportsFeature(featureName: Feature): boolean {
        return SUPPORTED_FEATURES.includes(featureName as any);
    }
```

Reply:
There is a case where `SUPPROTED_FEATURES` gets modified so we cannot define it as const, we can think about it separately as this shim should not be necessary anymore https://github.com/ioBroker/ioBroker.js-controller/blob/ef394c35acfd39727866bff82d6ceb852ccf3577/packages/adapter/src/lib/adapter/adapter.ts#L11825